rtb/norme_di_progetto_ver0.5.0.typ

```diff
@@ -175,6 +184,92 @@ Nella fase di implementazione, il fornitore deve stabilire o selezionare un mode
 === Analisi dei requisiti di sistema
 In questa fase, si analizzano le esigenze specifiche del sistema da sviluppare, documentando i requisiti funzionali, di sicurezza, di manutenzione, ergonomici e di interfaccia. Tali requisiti vengono valutati rispetto alla loro tracciabilità, consistenza con le esigenze iniziali, testabilità e fattibilità tecnica. Il risultato è una specifica dettagliata dei requisiti del sistema.
 
+==== Casi d'uso
+In questa fase rientra la realizzazione dei casi d'uso (_use case_): descrizioni formali di come un sistema _software_ interagisce con utenti o altre entità, chiamati attori, per raggiungere un obiettivo specifico. Questi sono composti da testo puro e sono usati per documentare le varie funzionalità attese dal sistema ad alto livello, quindi non specificando i dettagli implementativi. A un caso d'uso è preferibile abbinare anche il relativo diagramma #rifGlossario("UML").
+
+Ogni caso d'uso deve comprendere le seguenti informazioni:
+
+===== Nomenclatura
+Ogni caso d'uso deve essere identificato da un codice univoco, seguendo la nomenclatura: *UC[numero_use_case]*
+
+===== Attori
+Gli attori rappresentano agenti esterni che interagiscono con il sistema. Ogni caso d'uso specifica una funzionalità creata per un determinato attore.
+
+Esistono più tipi di attori:
+- *Attore primario*: L'utente o _stakeholder_ che avvia il caso d'uso ed ha un obbiettivo da raggiungere utilizzando il sistema.
+- *Attore di supporto*: Un utente o _stakeholder_ coinvolto nel caso d'uso, ma che non inizia l'interazione con il sistema. Può fornire ulteriori input o risorse necessarie per completare lo _use case_. La sua presenza non è obbligatoria all'interno di un caso d'uso.
+
+===== Precondizioni
+Condizioni che devono essere vere o soddisfatte affinché il caso d'uso possa iniziare. Definiscono lo stato iniziale del sistema ed i requisiti che devono essere garantiti affinché il caso d'uso sia valido.
+
+Esempi: 
+- Un utente deve essere autenticato nel sistema
+- Un prodotto deve essere aggiunto al carrello.
+
+
+===== Postcondizioni
+Stato finale del sistema dopo che il caso d'uso è stato completato. Utili per verificare che l'obiettivo sia stato raggiunto e per documentare le modifiche apportate al sistema.
+
+Esempio: Dopo un pagamento andato a buon fine, il sistema aggiorna lo stato dell'ordine a 'confermato' e invia una ricevuta via email.
+
+===== Trigger
+Il _trigger_ rappresenta l'evento o l'azione che scatena il caso d'uso. Questo può essere un'azione compiuta da un attore, come un _clic_ su un pulsante, oppure un evento esterno come il ricevimento di un messaggio da un altro sistema.
+
+Esempio: L'utente clicca sul pulsante 'Effettua pagamento'.
+
+===== Scenario principale
+Lo scenario principale descrive la sequenza di azioni _standard_ che l'attore ed il sistema eseguono per raggiungere l'obiettivo del caso d'uso. Include tutti i passi principali in ordine sequenziale.
+
+Esempio:
+1. L'utente inserisce i dati di accesso.
```

Comment:
Per quanto riguarda gli esempi, sempre una picolezza: starei sempre sullo stesso caso d'uso come esempio: in questo caso, per riferirci di più al nostro progetto, potresti fare la visualizzazione dashboard come esempio